controller/program/src/instructions/admin/remove_lst.rs

```diff
@@ -0,0 +1,170 @@
+        .and_then(TokenAccount::try_from_raw)
+        .map(|a| a.amount())
+        .ok_or(INVALID_ACCOUNT_DATA)?;
+    let protocol_fee_accumulator_balance =
+        RawTokenAccount::of_acc_data(abr.get(*accs.protocol_fee_accumulator()).data())
+            .and_then(TokenAccount::try_from_raw)
+            .map(|a| a.amount())
+            .ok_or(INVALID_ACCOUNT_DATA)?;
+
+    if lst_state.sol_value != 0 || lst_balance != 0 || protocol_fee_accumulator_balance != 0 {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::LstStillHasValue).into());
+    }
+
+    // Close protocol fee accumulator ATA
+    cpi.invoke_signed(
+        abr,
+        &token_prog,
+        CloseAccountIxData::as_buf(),
+        close_account_ix_account_handle_perms(
+            NewCloseAccountIxAccsBuilder::start()
+                .with_close(*accs.protocol_fee_accumulator())
+                .with_dst(*accs.refund_rent_to())
+                .with_auth(*accs.protocol_fee_accumulator_auth())
+                .build(),
+        ),
+        &[PdaSigner::new(&[
+            PdaSeed::new(&PROTOCOL_FEE_SEED),
+            PdaSeed::new(&[PROTOCOL_FEE_BUMP]),
+        ])],
+    )?;
+
+    // Close pool reserves ATA
+    cpi.invoke_signed(
+        abr,
+        &token_prog,
+        CloseAccountIxData::as_buf(),
+        close_account_ix_account_handle_perms(
+            NewCloseAccountIxAccsBuilder::start()
+                .with_close(*accs.pool_reserves())
+                .with_dst(*accs.refund_rent_to())
+                .with_auth(*accs.pool_state())
+                .build(),
+        ),
+        &[PdaSigner::new(&[
+            PdaSeed::new(&POOL_STATE_SEED),
+            PdaSeed::new(&[POOL_STATE_BUMP]),
+        ])],
+    )?;
+
+    // Shrink lst state list account  by 1 element,
+    // delete the account if it is now empty,
+    // and transfer any lamports excess of rent exemption to refund_rent_to
+    let lst_state_list_acc = abr.get_mut(*accs.lst_state_list());
+    let old_acc_len = lst_state_list_acc.data_len();
+    let byte_offset = lst_idx
+        .checked_mul(size_of::<LstStatePacked>())
+        .ok_or(INVALID_ACCOUNT_DATA)?;
+
+    lst_state_list_acc.data_mut().copy_within(
+        byte_offset + size_of::<LstStatePacked>()..old_acc_len,
+        byte_offset,
+    );
+    lst_state_list_acc.shrink_by(size_of::<LstStatePacked>())?;
+    let new_acc_len = lst_state_list_acc.data_len();
+
+    if new_acc_len == 0 {
+        abr.close(*accs.lst_state_list(), *accs.refund_rent_to())?;
+    } else {
+        let lamports_surplus = lst_state_list_acc
+            .lamports()
+            .checked_sub(Rent::get()?.min_balance(new_acc_len))
+            .ok_or(INVALID_ACCOUNT_DATA)?;
+        if lamports_surplus > 0 {
+            abr.transfer_direct(
+                *accs.lst_state_list(),
+                *accs.refund_rent_to(),
+                lamports_surplus,
+            )?;
+        }
```

Comment:
looks like `refund_excess_rent` can be a useful util fn too like the opposite of `pay_for_rent_exempt_shortfall`